crates/bevy_asset/src/path.rs

```diff
@@ -187,3 +187,9 @@ impl<'a> From<PathBuf> for AssetPath<'a> {
         }
     }
 }
+
+impl<'a> From<String> for AssetPath<'a> {
+    fn from(asset_path: String) -> Self {
+        asset_path.into()
+    }
+}
```

Comment:
This will stack overflow, as calling `asset_path.into()` will call `AssetPath::from(asset_path)`, which will call this trait method again. You should manually create the `AssetPath` struct, and do not make any calls to `from` and `into`.

Reply:
Alternatively...

```suggestion
impl<'a> From<String> for AssetPath<'a> {
    fn from(asset_path: String) -> Self {
        asset_path.as_str().into()
    }
}
```

Reply:
Alternatively...
```suggestion
impl<'a> From<String> for AssetPath<'a> {
    fn from(asset_path: String) -> Self {
        asset_path.as_str().into()
    }
}
```
As there is already existing logic for `&str`.

Reply:
Sorry. I overlooked this. I tried the solution with `as_str()` but it would not compile due to 
> cannot return value referencing function parameter `asset_path`
> returns a value referencing data owned by the current
So, I think the solution will be to create the AssetPath struct again?

Reply:
If that's the case then yes.